game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -2734,410 +2735,399 @@ public String allUnitStatsForExporter() {
-            .append(" moving per territory, ");
+        tuples.add(Tuple.of("Max " + getMovementLimit().getSecond() + " Units Moving per Territory",
+            String.valueOf(getMovementLimit().getFirst())));
       }
     }
+
     if (getAttackingLimit() != null) {
       if (getAttackingLimit().getFirst() == Integer.MAX_VALUE
           && (getIsAaForBombingThisUnitOnly() || getIsAaForCombatOnly())
           && !(Properties.getWW2V2(getData())
               || Properties.getWW2V3(getData())
               || Properties.getMultipleAaPerTerritory(getData()))) {
-        stats.append("max of 1 ").append(getAttackingLimit().getSecond()).append(" attacking per territory, ");
+        tuples.add(Tuple.of("Max " + getAttackingLimit().getSecond() + " Units Attacking per Territory", "1"));
       } else if (getAttackingLimit().getFirst() < 10000) {
-        stats.append("max of ").append(getAttackingLimit().getFirst()).append(" ")
-            .append(getAttackingLimit().getSecond()).append(" attacking per territory, ");
+        tuples.add(Tuple.of("Max " + getAttackingLimit().getSecond() + " Units Attacking per Territory",
+            String.valueOf(getAttackingLimit().getFirst())));
       }
     }
+
     if (getPlacementLimit() != null) {
       if (getPlacementLimit().getFirst() == Integer.MAX_VALUE
           && (getIsAaForBombingThisUnitOnly() || getIsAaForCombatOnly())
           && !(Properties.getWW2V2(getData())
               || Properties.getWW2V3(getData())
               || Properties.getMultipleAaPerTerritory(getData()))) {
-        stats.append("max of 1 ").append(getPlacementLimit().getSecond()).append(" placed per territory, ");
+        tuples.add(Tuple.of("Max " + getPlacementLimit().getSecond() + " Units Placed per Territory", "1"));
       } else if (getPlacementLimit().getFirst() < 10000) {
-        stats.append("max of ").append(getPlacementLimit().getFirst()).append(" ")
-            .append(getPlacementLimit().getSecond()).append(" placed per territory, ");
+        tuples.add(Tuple.of("Max " + getPlacementLimit().getSecond() + " Units Placed per Territory",
+            String.valueOf(getPlacementLimit().getFirst())));
       }
     }
-    if (stats.indexOf(", ") > -1) {
-      stats.delete(stats.lastIndexOf(", "), stats.length() - 1);
+
+    final StringBuilder result = new StringBuilder();
+    for (final Tuple<String, String> tuple : tuples) {
+      result.append(tuple.getFirst());
+      if (!tuple.getSecond().isEmpty()) {
+        result.append(": <b>").append(tuple.getSecond()).append("</b>");
+      }
+      result.append("<br />");
+    }
+    return result.toString();
+  }
+
+  private static <T extends DefaultNamed> void addIntegerMapDescription(final String key,
+      final IntegerMap<T> integerMap, final List<Tuple<String, String>> tuples) {
+    if (integerMap != null && integerMap.size() > 0) {
```

Comment:
I really like that you extracted these blocks, but I would nitpick that the control flow, the `if` statement should have stayed with the client code block. With this extraction, when reading the main block, you're going to have to jump down 300 lines to find out this helper method maybe was a no-op. Also to construct the entire control flow you need to consider all the helper methods.

Reply:
@DanVanAtta Funny enough that is what I did first and then didn't really like the repetitive if blocks 4 times in a row in the client block so moved it into the helper method as well. I think you could also argue to make the helper methods return the tuple rather than adding to the list of tuples to have a cleaner input/output for the method. If you did that though then you'd have another if check on if the result is present before adding it or add logic to skip empty tuples later on. I think both of those points are kind of a wash in this situation so would rather just move forward with this.

Reply:
Interesting points. I would tend to agree. 

(side note: FWIW, the cleaner input/ouput is a good step, but seems like that is the place to stop without finding a better solution)

Reply:
Sorry, that may not have been very clear.

- The refactoring logic you walk through sounds reasonable, it is an interesting conclusion.
- It seems a number of problems come up when we have the potentially empty return type. My 2 cents, keep the extracted functions as void, find a better way to solve further problems if the code could still use it (probably could).

In general though tend to agree we are at diminishing returns, it is pretty interesting your first approach took that path.